src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanContext.hpp

```diff
@@ -316,6 +320,29 @@ struct TScanContext final
                         m_operationType = OperationType::DeleteAll;
                     }
                 }
+                else if constexpr (std::is_same_v<T, std::shared_ptr<nlohmann::json>>)
+                {
+                    m_data = arg;
+                    auto message = std::get<std::shared_ptr<nlohmann::json>>(m_data);
+                    std::string updateValue = message->value("update", "");
+
+                    if (!updateValue.empty() && updateValue == "success")
+                    {
+                        m_messageType = MessageType::DatabaseTrigger;
+                        m_type = ScannerType::DatabaseUpdate;
+                        m_operationType = OperationType::Insert;
+                    }
+                    else
+                    {
+                        std::string rebootValue = message->value("reboot", "");
+                        if (!rebootValue.empty() && rebootValue == "success")
+                        {
+                            m_messageType = MessageType::DatabaseTrigger;
+                            m_type = ScannerType::DatabaseReset;
+                            m_operationType = OperationType::DeleteAll;
+                        }
+                    }
```

Comment:
There are some other comments about the approach in the PR.
If the action to delete absolutely everything is required, it shouldn't be handled as an event but as clean-before-start operation.

Reply:
Done in new refactor. We can discuss this again if further changes need to be implemented.